lib/core/public/teardown.js

```diff
@@ -0,0 +1,20 @@
+import cache from '../base/cache';
+
+/**
+ * Clean up axe-core tree and caches
+ */
+function teardown() {
```

Comment:
As above, I think we should assert here that _tree and _selectorData are defined. If someone calls teardown if there's nothing to tear down, something has gone wrong and we should let 'm know.

Reply:
I strongly agree that `setup()` should prevent you from calling it multiple times. However, I don't see the harm in calling `teardown()` more than once/when not "setup". Consider the following:

```js
try {
  axe.setup()
  axe.foo()
} catch {
  // Oh well, we tried!
}
axe.teardown()
```

or:

```js
function doSomeAxeThing() {
  // Ensure we empty out any previous state.
  axe.teardown()
  // Do the axe thing.
  return axe.bar()
}
```

Reply:
Since we're just cleaning up I also see no harm in calling it multiple times. However, since it's not exactly clear that `axe.run` teardowns for you, I'll add a comment so it's understood.

Reply:
Good point Stephen.